src/backend/optimizer/plan/planner.c

```diff
@@ -2579,6 +2579,26 @@ grouping_planner(PlannerInfo *root, double tuple_fraction)
 												  offset_est, count_est);
 			}
 
+			/*
+			 * Like pathlist:
+			 * Take Motion cost into accout before standard_planner().
+			 * Don't foo by one-phase LIMIT with partial_path here.
```

Comment:
Are these logics duplicated with the above logic? Should we consider moving the above logic directly here instead of adding once more?

Reply:
> Are these logics duplicated with the above logic?

Yeah, the codes are same.

> Should we consider moving the above logic directly here instead of adding once more?

No, the codes are same, but for different purpose and condition. 
The previous one are used for collect results for Limit  on QD, that must be ahead of add_limit_node.
And this new one is used for Motion cost before we do it finally at standard_planner, and we must put this one after Limit.
We couldn't combine them easily, it will introduce complex logic with `if conditions` for just saving a couple of lines.
Maybe I should tell more behind the comments: `Don't foo by one-phase LIMIT with partial_path here.`

Reply:
Oh, that make sense.